### PR TITLE
Modified SuspendedTabletsIT to confirm suspension cleared

### LINKED ISSUE #3314: Suspension may not be properly cleared for tablets in some cases
**Describe the bug**
While working on #3284 I noticed this code :

https://github.com/apache/accumulo/blob/15c9f55280f4ce28a80b800f1e05128f33c85db3/server/base/src/main/java/org/apache/accumulo/server/manager/state/MetaDataStateStore.java#L140-L151

Tracing back what calls this method, AFAICT its only passed TabletLocationState objects with a non-null suspend. If my analysis is correct then this method ignores all of its input.

Seems like the code should be :

```java
      for (TabletLocationState tls : tablets) {
        if (tls.suspend != null) {
          tabletsMutator.mutateTablet(tls.extent).deleteSuspension().mutate();

        }
     }
```

This need a bit more investigation.  Took a quick look at 1.10 code and it seems to be the same there.

**Expected behavior**
The suspension is properly cleared.

### PR DESCRIPTION
**Pull request by dlmarion**
Tablet suspension marker in Tablet metadata is cleared on a subsequent tablet unload

Closes #3314

### CODE REVIEW

File: test/src/main/java/org/apache/accumulo/test/manager/SuspendedTabletsIT.java [hunk 1]
```diff
@@ -322,6 +325,20 @@ private void suspensionTestBody(TServerKiller serverStopper) throws Exception {
         Thread.sleep(1000);
         ds = TabletLocations.retrieve(ctx, tableName);
       }
+
```
Comment (keith-turner) [inline test/src/main/java/org/apache/accumulo/test/manager/SuspendedTabletsIT.java:328]: I currently am in the process of trying to understand this test.  While doing so I added the following at this loacation and ran it locally.

```java
log.info("hc:{} sc:{} ac:{}", ds.hostedCount, ds.suspendedCount, ds.assignedCount);
```

I am seeing this output the following.  So even before the tablet server is stopped the suspend count is already zero.

```
hc:30 sc:0 ac:0
```

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/manager/SuspendedTabletsIT.java:328]: Interesting, I'll look into this

Comment (dlmarion) [reply test/src/main/java/org/apache/accumulo/test/manager/SuspendedTabletsIT.java:328]: I was able to reproduce your log information. Looking now at where `deleteSuspension` is called, it's being called in `MetadataStateStore.setFutureLocations` and `MetadataStateStore.setLocations`. The changes to this test likely don't need to be merged. Also, I think #3314 can be closed without changes as the code in the description is not reflective of the current state of the code now.

**Comment by dlmarion** [comment]
I talked with @keith-turner . He is working on a different PR for this.

**Comment by keith-turner** [comment]
OPened #4295.  Was finally able to figure out the mystery around #3314 because of [this comment](https://github.com/apache/accumulo/blob/f2adaebba28b5e47ea03a850254e901a0c7be371/server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java#L333) plus where the comment fell in the code.  At the place where comment fell a tablets goal state is !HOSTED and its current state is SUSPENDED.  So that lead me to suspect the intent of the code was to clear suspend markers for offline tables.

**Pull request closed without merging**